src/pages/CustomerInvoicePage.js

```diff
@@ -76,17 +76,13 @@ export const CustomerInvoicePage = ({ transaction, runWithLoadingIndicator, rout
   const onAddMasterList = () =>
     runWithLoadingIndicator(() => dispatch(PageActions.addMasterListItems('Transaction')));
 
-  const renderPageInfo = useCallback(
-    () => (
-      <PageInfo
-        columns={getPageInfoColumns(pageObject, dispatch, PageActions)}
-        isEditingDisabled={isFinalised}
-      />
-    ),
-    [comment, theirRef, isFinalised]
-  );
+  const pageInfoColumns = useCallback(getPageInfoColumns(pageObject, dispatch, PageActions), [
```

Comment:
Wondering if it might be worth adding some comments just indicating what method does and the purpose of the memoisation, e.g. 

```
 // Initialise column getter. Memoised to only run component mount.
 const pageInfoColumns = useCallback(getPageInfoColumns(pageObject, dispatch, PageActions), [
    comment,
    theirRef,
    isFinalised,
  ]);
```

Reply:
I think this sort of comment will get outdated quite quickly so i'm not 100% sold on the need? essentially it is: 

```
// This component is memoized and updated when this occurs
memoizeThisFunction(someFunction, [when this occurs])
```

Reply:
Arguably all comments will get out of date, but I still think they're worth doing.

If you're introducing features to a shared codebase, I think it makes sense to add minimal documentation indicating the fundamental life cycle of a component.

I wouldn't suggest commenting all `useCallback(args, ...)` because I think you're right that that could be a bit too much. But, I think `useCallback(args, [])` has a unique enough use-case (i.e. run only on mount) such that adding a comment could significantly reduce the time it would take a newbie to understand the code.

I agree that commenting too much isn't great, because of maintenance, and that ideally you want self-documenting code, but I'd still lean towards documenting key life cycle events of a component (i.e. ones which have clear analogues to life cycle methods).